keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -37,8 +37,12 @@ class GPT2CausalLM(Task):
     A causal language model (LM) predicts the next token based on previous
     tokens the next token based on previous tokens, which is the way GPT2 gets
     pretrained. You can finetune `GPT2CausalLM` to generate text similar to
-    the custom dataset. `GPT2CausalLM` also has a method `generate()`, which
-    generates text based on given prompt.
+    the custom dataset.
+
+    This model has a `generate()` method, which generates text based on a
+    prompt. The generation strategy used is controlled by an additional
+    `sampler` argument on `compile()`. You can recompile the model with
```

Comment:
I can check with @fchollet, but I actually think this may be a feature not a bug. Changing the sampler will mean you have to do this potentially quite slow compilation step when you generate next. The fact that users literally have to write `compile` is a good clue as to what is going on, and fits well with existing paradigms. It's very explicit.